FireboltNETSDK/Client/FireboltDataReader.cs

```diff
@@ -0,0 +1,473 @@
+        private string? _fullTableName;
+        private QueryResult _queryResult;
+        private int _depth;
+        private int _currentRowIndex = -1;
+        private static IDictionary<string, Type> typesMap = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "boolean", typeof(bool) },
+
+            { "byte", typeof(byte) },
+            { "short", typeof(short) },
+            { "int16", typeof(short) },
+            { "int", typeof(int) },
+            { "integer", typeof(int) },
+            { "int32", typeof(int) },
+            { "bigint", typeof(long) },
+            { "long", typeof(long) },
+            { "int64", typeof(long) },
+
+            { "float", typeof(float) },
+            { "double", typeof(double) },
+            { "double precision", typeof(double) },
+            { "decimal", typeof(decimal) },
+
+            { "string", typeof(string) },
+            { "text", typeof(string) },
+
+            { "date", typeof(DateTime) },
+            { "datetime", typeof(DateTime) },
+            { "timestamp", typeof(DateTime) },
+            { "timestamptz", typeof(DateTime) },
+            { "DateTime64", typeof(DateTime) },
+        };
+
+        public FireboltDataReader(string? fullTableName, QueryResult queryResult, int depth = 0)
+        {
+            _fullTableName = fullTableName;
+            _queryResult = queryResult;
+            _depth = depth;
+        }
+
+        public override object this[string name] { get => GetValue(GetOrdinal(name)); }
+
+        public override object this[int ordinal] { get => GetValue(ordinal); }
+
+        public override bool IsClosed { get => _closed; }
+
+        public override bool HasRows { get => _queryResult.Data.Count > 0; }
+
+        public override int FieldCount { get => _queryResult.Meta.Count; }
+
+        public override int Depth { get => _depth; }
+
+        public override int RecordsAffected { get => 0; }
+
+        public override int VisibleFieldCount { get => FieldCount; }
+
+        public override void Close()
+        {
+            _closed = true;
+        }
+
+        public override Task CloseAsync()
+        {
+            Close();
+            return Task.CompletedTask;
+
+        }
+
+        public override bool GetBoolean(int ordinal)
+        {
+            object value = GetValue(ordinal);
+            switch (value)
+            {
+                case bool b: return (bool)b;
+                case float f: return (float)f != 0;
+                case double d: return (double)d != 0;
+                case decimal d: return (decimal)d != 0;
+                case byte b: return (byte)b != 0;
+                case short s: return (short)s != 0;
+                case int i: return (int)i != 0;
+                case long l: return (long)l != 0;
+                case string s: return bool.Parse(s);
+                default: throw new InvalidCastException($"Cannot cast ({value.GetType()}){value} to boolean");
```

Comment:
As mentioned on CR, let's verify that this is the expected behaviour and we shouldn't just throw an error if the value is not bool.

Reply:
Yes, this is up to implementation. Please see the snapshot from the MySQL driver:

```
			return (boolVal == -1 || boolVal > 0);
		case Types.BIT:
		case Types.TINYINT:
		case Types.SMALLINT:
		case Types.INTEGER:
		case Types.BIGINT:
		case Types.DECIMAL:
		case Types.NUMERIC:
		case Types.REAL:
		case Types.FLOAT:
		case Types.DOUBLE:
			boolVal = getLong(columnIndex, false);

```

Reply:
I understood how, but I had questions to why. To me it sounds counterintuitive to request an incorrect type from the specific type returned from the database and get a result.
I'm not sure why you're using an example in different language and interface when there are opensource c# implementations for many databases. Here's postgres, which we're basing our SQL syntax on - https://github.com/npgsql/npgsql/blob/3d4b1d709982468345aca212eceee17e69869087/src/Npgsql/NpgsqlDataReader.cs#L1258

However, I admit it proves your point - Postgres also casts types to the requested type so I'm resolving this conversation.